src/ess/spectroscopy/indirect/time_of_flight.py

```diff
@@ -0,0 +1,162 @@
+    """Compute a lookup table for time-of-flight as a function of distance and
+    time-of-arrival.
+
+    This is a wrapper around :func:`ess.reduce.time_of_flight.compute_tof_lookup_table`
+    for indirect geometry spectrometers.
+    """
+    return reduce_time_of_flight.eto_to_tof.compute_tof_lookup_table(
+        simulation=simulation,
+        ltotal_range=LtotalRange(l1_range),
+        distance_resolution=distance_resolution,
+        time_resolution=time_resolution,
+        pulse_period=pulse_period,
+        pulse_stride=pulse_stride,
+        error_threshold=error_threshold,
+    )
+
+
+def detector_time_of_flight_data(
+    sample_data: DataAtSample[RunType],
+    lookup: TimeOfFlightLookupTable,
+    pulse_stride_offset: PulseStrideOffset,
+) -> DetectorTofData[RunType]:
+    """
+    Convert the time-of-arrival data to time-of-flight data using a lookup table.
+
+    The output data will have a time-of-flight coordinate.
+
+    This is a wrapper around
+    :func:`ess.reduce.time_of_flight.detector_time_of_flight_data`
+    for indirect geometry spectrometers.
+    """
+    result = reduce_time_of_flight.eto_to_tof.detector_time_of_flight_data(
+        detector_data=DetectorData[RunType](sample_data),
+        lookup=lookup,
+        ltotal=DetectorLtotal(sample_data.coords['L1']),
+        pulse_stride_offset=pulse_stride_offset,
+    )
+    # This is time-of-flight at the sample.
+    result.bins.coords['sample_tof'] = result.bins.coords.pop('tof')
+    del result.bins.coords['event_time_offset']
+    del result.bins.coords['event_time_zero']
+    return result
+
+
+def monitor_time_of_flight_data(
+    monitor_data: MonitorData[RunType, MonitorType],
+    lookup: TimeOfFlightLookupTable,
+    ltotal: MonitorLtotal[RunType, MonitorType],
+    pulse_stride_offset: PulseStrideOffset,
+) -> MonitorTofData[RunType, MonitorType]:
+    """
+    Convert the time-of-arrival data to time-of-flight data using a lookup table.
+
+    The output data will have a time-of-flight coordinate.
+
+    This is a wrapper around
+    :func:`ess.reduce.time_of_flight.monitor_time_of_flight_data`
+    for indirect geometry spectrometers.
+    """
+    result = reduce_time_of_flight.eto_to_tof.monitor_time_of_flight_data(
+        monitor_data=monitor_data.rename(t='tof'),
+        lookup=lookup,
+        ltotal=ltotal,
+        pulse_stride_offset=pulse_stride_offset,
+    )
+    return result
+
+
+def compute_monitor_ltotal(
+    monitor_data: MonitorData[RunType, MonitorType],
+    coord_transform_graph: MonitorCoordTransformGraph,
+) -> MonitorLtotal[RunType, MonitorType]:
+    """Compute the path length from the source to the monitor."""
+    return MonitorLtotal[RunType, MonitorType](
+        monitor_data.transform_coords(
+            'Ltotal',
+            graph=coord_transform_graph,
+            keep_intermediate=False,
+            keep_aliases=False,
+            rename_dims=False,
+        ).coords['Ltotal']
+    )
```

Comment:
Where is this needed? Is it not part of the `GenericTofWorkflow` (if so, why not?)?

Reply:
I wrote this so that it uses the custom coord transform graph. Currently, this is equivalent. But Greg's original workflow took the curvature of the guide into account and I wanted to have to option to add that back in the future.

Reply:
Maybe the version in essreduce should allow for setting a custom graph, i.e., can we move this function there?

Reply:
I can try it. But I would also like to get this PR here merged to get the nightly tests (ours and DMSC) to work again. So can we move it in a separate step?